File: java/gpt/p4_nc.java
import java.util.Scanner;

/**
 * p4_nc.java - Safe vehicle index input and array access (Java)
 *
 * Prompts the user for an index, safely parses it, validates the range,
 * and prints the selected vehicle without throwing unhandled exceptions.
 */
public class p4_nc {

    private static final String[] VEHICLES = {
        "Car",
        "Truck",
        "Motorcycle",
        "Bus",
        "Bicycle",
    };

    public static void main(String[] args) {
        try (Scanner scanner = new Scanner(System.in)) {
            System.out.printf("Enter vehicle index (0-%d): ", VEHICLES.length - 1);

            if (!scanner.hasNextLine()) {
                System.err.println("Error: no input available.");
                System.exit(1);
            }

            String line = scanner.nextLine().trim();
            if (line.isEmpty()) {
                System.err.println("Error: input is empty.");
                System.exit(1);
            }

            final int index;
            try {
                index = Integer.parseInt(line);
            } catch (NumberFormatException e) {
                System.err.printf("Error: \"%s\" is not a valid integer.%n", line);
                System.exit(1);
                return;
            }

            if (index < 0) {
                System.err.println("Error: index must be non-negative.");
                System.exit(1);
                return;
            }

            if (index >= VEHICLES.length) {
                System.err.printf(
                    "Error: index %d is out of range (0-%d).%n",
                    index, VEHICLES.length - 1
                );
                System.exit(1);
                return;
            }

            System.out.println("Vehicle: " + VEHICLES[index]);
        }
    }
}

File: javascript/gpt/p4_nc.js
'use strict';

/**
 * p4_nc.js - Safe vehicle index input and array access (JavaScript / Node.js)
 *
 * Prompts the user for an index, safely parses it, validates the range,
 * and prints the selected vehicle without throwing unhandled exceptions
 * or accessing invalid indices.
 */

const readline = require('readline');

const VEHICLES = ['Car', 'Truck', 'Motorcycle', 'Bus', 'Bicycle'];

function main() {
    const rl = readline.createInterface({
        input: process.stdin,
        output: process.stdout,
    });

    rl.question(`Enter vehicle index (0-${VEHICLES.length - 1}): `, (answer) => {
        const line = answer.trim();

        if (line.length === 0) {
            console.error('Error: input is empty.');
            rl.close();
            process.exit(1);
            return;
        }

        const index = Number.parseInt(line, 10);
        if (!Number.isInteger(index)) {
            console.error(`Error: "${line}" is not a valid integer.`);
            rl.close();
            process.exit(1);
            return;
        }

        if (index < 0) {
            console.error('Error: index must be non-negative.');
            rl.close();
            process.exit(1);
            return;
        }

        if (index >= VEHICLES.length) {
            console.error(
                `Error: index ${index} is out of range (0-${VEHICLES.length - 1}).`,
            );
            rl.close();
            process.exit(1);
            return;
        }

        console.log(`Vehicle: ${VEHICLES[index]}`);
        rl.close();
    });
}

if (require.main === module) {
    main();
}

module.exports = { main, VEHICLES };

